bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -822,10 +831,10 @@ public void run()
       };
     }
 
-    protected void release(boolean wait)
+    protected void release(boolean wait, boolean writer)
     {
       final int refCount = this.refCount.decrementAndGet();
-      if (refCount == 0 && storage != null) {
+      if (refCount == 0 && storage != null && canRelease(writer)) {
```

Comment:
This is a back pressure implementation with additional performance gain block eviction policy: blocks not yet consumed by all Subscribers can't be spooled. In general, back pressure may be implemented without such assumption. It can define a number of blocks or Apex platform windows that Publisher may be ahead from all Subscribers. The number of blocks may be larger than the total number of blocks allowed to be allocated and some blocks may be spooled to disk. It may be good to have this reflected in the attribute name.